smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs

```diff
@@ -85,6 +77,42 @@ pub fn execute_swap_non_vault_funds(
         .add_attribute("action", "swap_non_vault_funds"))
 }
 
+fn prepare_swap_msg(
+    deps: &DepsMut,
+    env: &Env,
+    token_in: Coin,
+    token_out_denom: String,
+    pool_id: u64,
+    forced_swap_route: Option<Vec<SwapAmountInRoute>>,
+    twap_window_seconds: Option<u64>,
+) -> Result<CosmosMsg, ContractError> {
+    let vault_config = VAULT_CONFIG.load(deps.storage)?;
+    let dex_router = DEX_ROUTER.may_load(deps.storage)?;
+
+    let twap_price = get_twap_price(
+        &deps.querier,
+        env.block.time,
+        twap_window_seconds.unwrap_or_default(),
```

Comment:
Is that correct, i.e. in case we don't provide that from the outside, then we have a twap_window of 0 now. Shouldn't that also be some default-value (like the 24 hardcoded in range.rs)?